docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_testing-in-testing-farm.adoc

```diff
@@ -0,0 +1,51 @@
+= Adding an integration test
+
+link:https://docs.testing-farm.io/[Testing Farm] is a third-party test service that can be used as a test execution back-end for {ProductName}. This document guides you through an example of how to trigger Testing Farm tests from your {ProductName} *Integration tests*.
+
+.Prerequisites
+
+* You have created an application in {ProductName}.
+* You have a Testing Farm API key, which can be acquired by following the link:https://docs.testing-farm.io/Testing%20Farm/0.1/onboarding.html[Testing Farm onboarding guide].
+* You have have or are ready to create a git repository with link:https://fmf.readthedocs.io/[fmf] files suitable for use with the link:https://tmt.readthedocs.io/[tmt] testing tool. This is the testing technology used by Testing Farm under the hood. If you don't have a fmf tests already prepared, you can use our tmt link:https://github.com/ralphbean/tmt-hello-world[hello world repository].
+
+.Procedure
+
+You need to perform two major steps. Upload your Testing Farm API key, and register a *Integration test* to send requests to the Testing Farm API. Complete the following steps in the {ProductName} console:
+
+.Procedure - Upload your Testing Farm API key
+
+Follow similar instructions as in the xref:how-to-guides/configuring-builds/proc_creating-secrets-for-your-builds.adoc[Creating secrets for your builds] guide.
+
+. In {ProductName}, from the left navigation menu, select **Secrets**.
+. From the **Secrets** page, click **Add secret**.
+. From the **Add secret** page, choose **Build**.
+. Select secret type **Key/value secret**
+. For **Secret name**, enter `testing-farm-secret`.
+. Under **Key/value secret**, expand **Key/value 1**, then enter the key name `testing-farm-token`.
+NOTE: Be sure that the secret name is `testing-farm-secret` and the key name is `testing-farm-token`. The link:https://github.com/ralphbean/testing-farm-tekton/blob/main/tasks/testing-farm.yaml[tekton task] expects the secret and key name to be named these strings exactly.
+. For **Upload the file with value for your key or paste its contents**, paste the value of your token into the space under **Upload**.
+. Optional: Under **Labels**, add a label to tag or provide more context for your secret.
+. Click **Add secret**.
+
+.Procedure - Registering the Integration Test
+
+. Open an existing application and go to the *Integration tests* tab.
+. Select *Add integration test*.
+. In the *Integration test name* field, enter a name of your choosing to reflect your tests. The name `testing-farm` might be suitable.
+. In the *GitHub URL* field, enter `https://github.com/ralphbean/testing-farm-tekton`, which is a git repository containing a tekton pipeline and task that can send requests to the Testing Farm API.
```

Comment:
Maybe we should move this to github.com/redhat-appstudio?

On the other hand, maybe it's best to imply that it _isn't_ supported by keeping it under my namespace?

We could ask https://github.com/testing-farm if they're interested in keeping it under their org.

Reply:
https://github.com/testing-farm  this sounds like a good place to me. Till that we can keep it in your fork I'd say.

Reply:
Will there be multiple types of integration tests that we provide pipelines for (in an unsupported way)? I am not sure about submitting to testing-farm because of the dependence on a Snapshot... that is definitely a concept that we introduce.

Reply:
Yeah, I can imagine openrefactory providing a pipeline and rapidast providing a pipeline. You're right - it's a pipeline _for_ RHTAP because of the Snapshot reference.

Reply:
If that is the case, then it might be good to create a dedicated repository to host these types of pipeline definitions. We can clearly indicate in the repo that it is not supported in the same way (if desired). If not that, then a separate and related org can be created to house repos for each of these ITS types.